### PR TITLE
[Spree 2.1] Add 'pt' to list of available locales in test suite

### LINKED ISSUE #4835: [Spree 2.1] Available/default locale errors
There are several errors in the build related to available locales.

Examples:

```
  86) Spree::UserMailer#signup_confirmation user locale sends email in user locale when user locale is defined
      Failure/Error: I18n.default_locale = 'pt'
      
      I18n::InvalidLocale:
        "pt" is not a valid locale
      # ./spec/mailers/user_mailer_spec.rb:29:in `block (4 levels) in <top (required)>'

  87) Spree::UserMailer#signup_confirmation user locale sends email in default locale when user locale is not available
      Failure/Error: I18n.default_locale = 'pt'
      
      I18n::InvalidLocale:
        "pt" is not a valid locale
      # ./spec/mailers/user_mailer_spec.rb:29:in `block (4 levels) in <top (required)>'

...

  89) UserRegistrationsController via ajax sets user.locale from cookie on create
      Failure/Error: I18n.locale = spree_current_user.andand.locale || cookies[:locale] || I18n.default_locale
      
      I18n::InvalidLocale:
        "pt" is not a valid locale
      # ./app/helpers/i18n_helper.rb:14:in `set_locale'
      # ./spec/controllers/user_registrations_controller_spec.rb:56:in `block (3 levels) in <top (required)>'
```

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
Closes #4835 

Fixes several errors such as:

```
104) UserRegistrationsController via ajax sets user.locale from cookie on create
       Failure/Error: I18n.locale = spree_current_user.andand.locale || cookies[:locale] || I18n.default_locale

       I18n::InvalidLocale:
         "pt" is not a valid locale
       # ./app/helpers/i18n_helper.rb:14:in `set_locale'
       # ./spec/controllers/user_registrations_controller_spec.rb:56:in `block (3 levels) in <top (required)>'
```

### CODE REVIEW

**Comment by Matt-Yorkley** [comment]
Note: this PR has been updated slightly since it's last inclusion in `3-0-latest`, to fix error: 

```
it 'has two locales available'
```

It now checks for 3 and passes.

File: config/environments/test.rb [hunk 1]
```diff
@@ -35,7 +35,7 @@
 
   # Tests assume English text on the site.
   config.i18n.default_locale = "en"
-  config.i18n.available_locales = ['en', 'es']
+  config.i18n.available_locales = ['en', 'es', 'pt']
```
Comment (luisramos0) [inline config/environments/test.rb]: actually we need 'de' here as well to fix spec/features/admin/order_cycles_spec.rb:123:
:-)

Comment (sauloperez) [reply config/environments/test.rb]: I leave that one :point_up: to you @Matt-Yorkley. Feel free to merge afterwards.

Comment (Matt-Yorkley) [reply config/environments/test.rb]: Isso mudou... 

I changed that test in `spec/features/admin/order_cycles_spec.rb:123` from `:de` to `:pt` and now it's green.

Comment (luisramos0) [reply config/environments/test.rb]: :+1:

**Pull request merged**